README.md

```diff
@@ -1,294 +1,303 @@
+
+- [ ] Now when you refresh the page, it looks as though you are logged in. Next, try clicking the logout button.
+
+* **Question:** When you click "Logout", nothing happens unless you refresh the page. Why not?
+
+* **Your Answer:**
+
+---The navigation is still showing as though i am authenticated. My browser's local storage still contains the `journal-app` token and the app component's `currentUserId` in state has not been reset/set to null.
+
+- [ ] Update the `logout()` method to appropriately logout the user.
+
+* **Question:** What did you have to do to get the `logout()` function to work? Why?
+
+* **Your Answer:**
+
+--- I had to create the function in App.js and then pass it through each component so that when the logout component is clicked, the logout function is called and the token data is removed and the app component state's currentuserid is set to null
+
+- [ ] Following the patterns we used above, build the Signup feature.
+
+---
+
+- [ ] When a user logs in or signs up, we should bring them to the `/users` route. Update both features so that the user is moved to that route after a successful login/signup.
+
+---
+
+- [ ] Try logging out and then go directly to the `/users` route.
+
+* **Question:** What happens? What _should_ happen?
+
+* **Answer:**
+
+---I see the `/users` page as if i was authenticated. Because i am not authenticated, i should not be authorized to see this page.
+
+- [ ] Try _replacing_ the `/users` Route in `App.js` with the following:
+  ```jsx
+  <Route path='/users' render={() => {
+    return this.state.currentUserId ? <UsersContainer /> : <Redirect to='/login' />
+  }} />
+  ```
+
+* **Question:** Describe what is happening in the code above.
+
+* **Your Answer:**
+
+---Upon accessing `/users` route, the App component's state is checked for currentUserId. If currentUserID is present,  the app will render the UsersContainer component. If it is not present, the login component is instead rendered as the user is not authenticated, and thus not authorized to view the page.
+
+- [ ] Now try logging in. Then, when you're on the `/users` page, refresh the page.
+
+* **Question:** What happens and why?
+
+* **Your Answer:**
+
+--- I am not sure i fully grasp this quite yet, but i believe it is because the page is refreshed on the `/users` route before the app component is mounted and currentUserID is not yet set into state. because of this, the users page cannot be shown to the user as per the route, we are redirecting to login page when no currentUserID is found. it seems that the `/users` route execution precedes `componentDidMount()`. Is this correct?
+
+- [ ] To solve this problem, let's add a `loading` key to our App's state, with the default value set to `true`. When `componentDidMount()` finishes, set the `loading` key to equal `false`. Using this key, solve the issue of refreshing on the `/users` page. Make sure everyting continues to work whether you are logged in or out.
+

```

Comment:
The app component is mounted, but still needs to contact the backend server and wait for a response to make sure the token is valid and retrieves user and post info. But the users route will render immediately if it sees a user id in state, which it does since one was already stored in local storage. So, we need to delay the render until we have successfully retrieved data from the backend server.